### PR TITLE
An attempt at making Recursive Loop Of Complex Types an opt-in feature 

### PR DESCRIPTION
**Pull request by FortuneN**
This would allow the brave/stupid like me to opt into recursions and be prepared to deal with the fallout
#33

I tried this on a project i'm currently working on and so far no problems.

I'm prepared to adjust this PR as necessary to meet your standards.

### CODE REVIEW

File: Community.Data.OData.Linq/OdataLinqExtensions.cs [hunk 1]
```diff
@@ -54,13 +54,16 @@ public static ODataQuery<T> OData<T>(this IQueryable<T> query, Action<ODataSetti
...
+            configuration?.Invoke(settings);
+
             if (edmModel == null)
             {
                 edmModel = Models.GetOrAdd(typeof(T), t =>
                 {
                     ODataConventionModelBuilder builder = new ODataConventionModelBuilder();
                     builder.AddEntityType(t);
-                    builder.AddEntitySet(t.Name, new EntityTypeConfiguration(new ODataModelBuilder(), t));
+                    builder.AddEntitySet(t.Name, new EntityTypeConfiguration(new ODataModelBuilder { AllowRecursiveLoopOfComplexTypes = settings.AllowRecursiveLoopOfComplexTypes }, t));
```
Comment (IharYakimush) [inline Community.Data.OData.Linq/OdataLinqExtensions.cs]: add AllowRecursiveLoopOfComplexTypes property to hash calculation function below

**Comment by IharYakimush** [comment]
Hi @FortuneN please add unit tests and resolve my comment related to hash calculation

File: Community.Data.OData.Linq/OdataLinqExtensions.cs (hunk 1 above)
Comment (FortuneN) [reply Community.Data.OData.Linq/OdataLinqExtensions.cs]: Done.

**Comment by FortuneN** [comment]
I changed the manner in which i access the ODataSettings in StructuralTypeConfiguration.
It seems a bit better than just transferring one property of the settings.
This way, all other settings values are usable (if need be) at that level.

**Comment by FortuneN** [comment]
I changed the manner in which i access the the ODataSettings in StructuralTypeConfiguration.
It seems a bit better than just transferring just

**Review by IharYakimush — APPROVED** [review]

**Comment by IharYakimush** [comment]
Hi @FortuneN, this change available in Community.OData.Linq.2.1.24-rc

**Pull request merged**